mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowHttpException.java

```diff
@@ -0,0 +1,51 @@
+package org.mlflow.tracking;
+
+/**
+ * HTTP exception.
+ */
+public class MlflowHttpException extends MlflowClientException {
+
+  public MlflowHttpException(int statusCode, String reasonPhrase) {
+    super("statusCode=" + statusCode + " reasonPhrase=" + reasonPhrase);
+    this.statusCode = statusCode;
+    this.reasonPhrase = reasonPhrase;
+  }
+
+  public MlflowHttpException(int statusCode, String reasonPhrase, String bodyMessage) {
+    super("statusCode=" + statusCode + " reasonPhrase=[" + reasonPhrase + "] bodyMessage=["
+      + bodyMessage + "]");
+    this.statusCode = statusCode;
+    this.reasonPhrase = reasonPhrase;
+    this.bodyMessage = bodyMessage;
+  }
+
+  private int statusCode;
+
+  public int getStatusCode() {
+    return statusCode;
+  }
+
+  public void setStatusCode(int statusCode) {
```

Comment:
Do we need setters on an Exception? Seems better to leave them out.

Reply:
We want to give our users the ability to set their own destiny.

Reply:
(Removed...)